### PR TITLE
planner: forbid point get with select lock choose tiflash as datasource path

### LINKED ISSUE #39344: point get for update reading from tiflash
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
use test;
drop table if exists t;
create table t(a int, b int, c int, primary key(a, b));
insert into t values(1,2,3), (4,5,6);
alter table t set tiflash replica 1;
select sleep(10);
set tidb_enable_tiflash_read_for_write_stmt = on;
set tidb_isolation_read_engines='tidb, tiflash';

begin;
explain select a, b from t where a = 1 and b = 2 for update;
```
### 2. What did you expect to see? (Required)
```
MySQL [test]> explain select a, b from t where a = 1 and b = 2 for update;
(1815, "Internal : No access path for table 't' is found with 'tidb_isolation_read_engines' = 'tidb,tiflash', valid values can be 'tikv, tiflash'.")
```
### 3. What did you see instead (Required)
```
MySQL [test]> begin;
Query OK, 0 rows affected (0.00 sec)

MySQL [test]> explain select a, b from t where a = 1 and b = 2 for update;
+--------------------------+---------+--------------+---------------+-------------------------------------------------+
| id                       | estRows | task         | access object | operator info                                   |
+--------------------------+---------+--------------+---------------+-------------------------------------------------+
| SelectLock_6             | 0.01    | root         |               | for update 0                                    |
| └─TableReader_8          | 1.00    | root         |               | data:TableRangeScan_7                           |
|   └─TableRangeScan_7     | 1.00    | cop[tiflash] | table:t       | range:[1 2,1 2], keep order:false, stats:pseudo |
+--------------------------+---------+--------------+---------------+-------------------------------------------------+
3 rows in set (0.00 sec)
```
### 4. What is your TiDB version? (Required)

```
Release Version: v6.5.0-alpha
Edition: Community
Git Commit Hash: 8fc4535575b7cb8b3ed0ef9fe042b510815f423d
Git Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time: 2022-11-22 14:33:54
GoVersion: go1.19.2
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
```

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close #39344 

Problem Summary:

### What is changed and how it works?

forbid point get with select lock choose tiflash as datasource path

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — COMMENTED** [review]
We can do this just in the plan builder?

File: planner/property/physical_property.go [hunk 1]
```diff
@@ -202,6 +202,9 @@ type PhysicalProperty struct {
 	// RejectSort means rejecting the sort property from its children, but it only works for MPP tasks.
 	// Non-MPP tasks do not care about it.
 	RejectSort bool
+
+	// InLock indicates the Plan is under LogicalLock
+	InLock bool
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply planner/property/physical_property.go]: updated, use `StmtCtx.LockTableIDs`  instead

**Comment by Yisaer** [comment]
> We can do this just in the plan builder?

i tended to check it during `findbestTask` thus we can directly used the info in the `StmtCtx.LockTableIDs`

**Review by qw4990 — APPROVED** [review]

**Review by fixdb — APPROVED** [review]
+1

File: planner/core/integration_test.go [hunk 2]
```diff
@@ -7554,6 +7554,26 @@ func TestEnableTiFlashReadForWriteStmt(t *testing.T) {
 	checkMpp(rs)
 }
 
+func TestPointGetWithSelectLock(t *testing.T) {
```
Comment (gengliqi) [inline planner/core/integration_test.go:7557]: Not just a test for a primary key, a test for a unique key is also necessary.

Comment (Yisaer) [reply planner/core/integration_test.go:7557]: updated

**Review by gengliqi — APPROVED** [review]
LGTM

**Pull request merged**